### PR TITLE
feat: [DHIS2-11402] navigate to enrollment edit event page

### PR DESCRIPTION
**Pull request by simonadomnisoru**
Navigate to the edit event page when clicking an event in the "stages and events" Widget. The PR is built on the work done in the `WidgetStagesAndEvents` https://github.com/dhis2/capture-app/tree/DHIS2-10222

### CODE REVIEW

**Review by huocha — APPROVED** [review]
LGTM

**Review by JoakimSM — COMMENTED** [review]
Thanks for this PR @simonadomnisoru.

May I suggest that we add `cursor: 'pointer` to the row style? (At least for now, until they come to a conclusion here: https://github.com/dhis2/notes/discussions/297)

File: src/core_modules/capture-core/components/WidgetStagesAndEvents/Stages/Stage/StageDetail/StageDetail.component.js [hunk 1]
```diff
@@ -100,6 +100,7 @@ const StageDetailPlain = ({ events, eventName, dataElements, classes }: Props) =
                     .map(id => (
                         <DataTableCell
                             key={id}
+                            onClick={() => onEventClick(events[index].event, events[index].programStage)}
```
[the remark this answers is not in the document]
Comment (huocha) [reply src/core_modules/capture-core/components/WidgetStagesAndEvents/Stages/Stage/StageDetail/StageDetail.component.js:103]: @JoakimSM I added `id` as eventId in the dataSource in the main ticket

**Comment by simonadomnisoru** [comment]
I had to resolve conflicts with https://github.com/dhis2/capture-app/tree/DHIS2-10222 and rebase this branch.

**Pull request closed without merging**